src/haxelib/Util.hx

```diff
@@ -0,0 +1,60 @@
+package haxelib;
+
+#if macro
+import haxe.macro.Expr;
+
+using haxe.macro.Tools;
+#end
+
+using StringTools;
+
+class Util {
+	macro static public function rethrow(e) {
+		return if (haxe.macro.Context.defined("neko"))
+			macro neko.Lib.rethrow(e);
+		else
+			macro throw e;
+	}
+
+	#if macro
+	static function readVersionFromHaxelibJson() {
+		return haxe.Json.parse(sys.io.File.getContent("haxelib.json")).version;
+	}
+	#end
+
+	macro static public function getHaxelibVersion() {
+		return macro $v{readVersionFromHaxelibJson()};
+	}
+
+	macro static public function getHaxelibVersionLong() {
+		var version:String = readVersionFromHaxelibJson();
+		var p;
+		try {
+			//check if the .git folder exist
+			//prevent getting the git info of a parent directory
+			if (!sys.FileSystem.isDirectory(".git"))
+				throw "Not a git repo.";
```

Comment:
Should this really throw things in macro context?

Reply:
I didn't write this function, I just moved it into this package... but I guess it would be worth checking if it doesn't cause issues if someone uses haxelib as a library.

Reply:
Though, is it an issue if it gets caught right after on line 55?

Reply:
Hmm, right, though it's strange that the caught exception there is then ignored... pretty wacky control flow management, but oh well.

Reply:
Yh, when it's not a git repository it reverts to what's found in `haxelib.json`. I agree it could probably do with being rewritten but probably better for another PR x)